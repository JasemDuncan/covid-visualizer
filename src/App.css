.contenedor {
  border-collapse: collapse;
  width: 100%;  
  margin: 0px;
  padding: 0px;
}

body{
  background-color: rgb(241, 77, 151);
  font-family: 'Lato', sans-serif;
  font-family: 'Montserrat', sans-serif;

}

.elemento:nth-child(even) {background-color: rgb(207, 67, 118);}
.elemento:nth-child(odd) {
  background-color: rgb(206, 48, 106);  
}

a {
  color: rgb(255, 255, 255);
  text-decoration: none; /* no underline */
}

.countryName{
  color: white;
  font-size: 13px;
  font-weight:700;
  text-transform: uppercase;
}
.confirmedSubtitle{
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-end;
  color: white;
  font-size: 10px;
  font-weight:500;  
}

.confirmedSubtitleDetail{
  color: white;
  font-size: 10px;
  font-weight:bold;  
}

.allCountry{
  height: 120px;
}

.contenedor {
  margin: 0;
  padding: 0;
}

.statsByCountry{
  padding-left: 10px;
  font-size: 11px;
  font-weight:700; 
  color: white;
  background-color: rgb(207, 67, 118);
  text-transform: uppercase;
}

.title {
  font-size: 15px;
  font-weight:bold; 
  color: white;  
}

.sandwichMenu {
  font-style: 11px;
  padding: 0px;
  margin: 0px;
}

.searchInput{
  padding-left: 10px;
  background-color: rgb(202, 98, 136);
  width: 100%;
  color: white;
  border: none ;
  overflow: auto;
  outline: none;
  box-shadow: none;  
}

::placeholder {
  color: rgb(255, 255, 255);
  opacity: 0.8; /* Firefox */
  font-size: 10px;
  font-weight:700; 
}

.imgContainer{    
  height: 150px;      
  background-color: rgb(206, 48, 106);
}

.imgGif{    
  height: 100%;
  object-fit: none;  
  opacity: 0.07;  
  background-repeat: repeat;
}

.ontop{
  color: white;      
  display: flex;
  flex-direction: column;    
  justify-content: center;
  align-items: flex-end;  
}

.titleWorld{
  font-size: 18px;
  font-weight:bold;    
  text-transform: uppercase;
}

.subtitleWorld{
  font-size: 12px;
  font-weight:500;    
}

.stadisticsWorld{
  font-size: 15px;
  font-weight:400;   
}

/* All for detail info */

.contenedorInfo {  
  width: 100%;  
  margin: 0px;
  padding: 0px;    
  background-color: rgb(206, 48, 106);  
}

.contenedorInfoTwo {  
  width: 100%;  
  margin: 0px;
  padding: 0px;    
  background-color: rgb(207, 67, 118);  
}

.elementoInfo {  
  width: 100%;
  height: 60px;
  justify-content: space-between;
  align-items: center;
}

.confirmedInfo{
  color: white;
  font-size: 15px;
  font-weight:500;  
}

.confirmedInfo::first-letter {
  font-size: 16px;
  font-weight: bold;  
}

.detailInfo {
  width: 160px;  
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  
}

.confirmedSubtitleInfo{
  display: flex;
  flex-direction: row;  
  justify-content: space-between;
  width: 150px;  
  color: white;
  font-size: 14px;
  font-weight: 600;   
}

.iconInfo {
  display: flex;  
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80px;
  /* background-color: chartreuse; */
}