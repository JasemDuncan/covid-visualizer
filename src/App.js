import './App.css';
import React from 'react';
import {
  BrowserRouter as Router,
  Routes,
  Route,
  NavLink,
} from 'react-router-dom';
import { useSelector } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faChevronLeft,
  faMicrophone, faCog, faSignal,
} from '@fortawesome/free-solid-svg-icons';
import Countries from './components/Countries';
import Info from './components/Info';

function App() {
  const countries = useSelector((state) => state.countriesReducer);
  return (
    <Router>
      <>
        <div>
          <div className="row d-flex justify-content-center align-items-center mt-3">
            <div className="col-2 d-flex justify-content-center align-items-center">
              <NavLink to="/">
                <FontAwesomeIcon icon={faChevronLeft} size="sm" color="white" />
              </NavLink>
            </div>
            <div className="col-7 d-flex justify-content-center align-items-center">
              <div className="title">COVID VISUALIZER â„¢</div>
            </div>
            <div className="col-3 d-flex justify-content-around align-items-center">
              <FontAwesomeIcon icon={faSignal} size="sm" color="yellow" />
              <FontAwesomeIcon icon={faMicrophone} size="sm" color="white" />
              <FontAwesomeIcon icon={faCog} size="sm" color="white" />
            </div>
          </div>
          <Routes>
            <Route path="/" element={<Countries />} />
            { countries.map((country) => (
              <>
                <Route
                  key={country.id}
                  path={`/${country.id.includes(' ') ? country.id.replace(/\s/g, '-') : country.id}`}
                  element={<Info country={country} />}
                />
              </>

            ))}
          </Routes>

        </div>
      </>
    </Router>
  );
}

export default App;
