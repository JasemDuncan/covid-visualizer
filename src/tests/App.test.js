import React from 'react';
import '@testing-library/jest-dom';
import {
  render, fireEvent, screen, cleanup,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import App from '../App';
import store from '../redux/configurationStore';

afterEach(() => {
  cleanup();
});

describe('App test', () => {
  // const initialState = { data };
  // const mockStore = configureStore();
  // let store;

  // it('Render Correctly', () => {
  //   // store = mockStore(initialState);
  //   // const tree = renderer.create(
  //   //   <Provider store={store}>
  //   //     <App />
  //   //   </Provider>,
  //   // );
  //   // expect(tree).toMatchSnapshot();
  // });
  test('Testing App component', () => {
    const app = renderer.create(<Provider store={store}><App /></Provider>).toJSON();
    expect(app).toMatchSnapshot();
  });

  test('Testing App component', () => {
    const app = renderer.create(<Provider store={store}><App /></Provider>).toJSON();
    expect(app.children).not.toBeNull();
  });
});
