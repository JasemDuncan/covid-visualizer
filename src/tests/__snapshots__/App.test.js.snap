// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test Testing App component 1`] = `
<div>
  <div>
    <div>
      <img
        alt="logo"
        height="40"
        src="logo.svg"
      />
      <h1>
        COVID VISUALIZER
      </h1>
    </div>
    <div>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
         Country
      </a>
    </div>
    <div>
      <div>
        Conf.
      </div>
    </div>
  </div>
  <hr />
  <div>
    <div>
      <input
        onChange={[Function]}
        placeholder="search"
        type="text"
      />
    </div>
  </div>
</div>
`;
